### DISCUSSION TITLE
How to implement Opencvsharp in Avaloniaui to display camera video data on the interface.（WriteableBitmapConverter）

### DISCUSSION THREAD

**Discussion by qq908382818** [AvaloniaUI/Avalonia#15234] [General]
How to implement Opencvsharp in Avaloniaui to display camera video data on the interface. Under WIN, it can be displayed on the picturebox normally. How to implement the method WriteableBitmapConverter. Method OpenCvSharp on Win There is a ready-made conversion from Mat to Bitmap, but AVA has not been implemented. Find code；thanks!

**Comment by maxkatz6** [comment]
See this discussion https://github.com/AvaloniaUI/Avalonia/discussions/11074

**Reply by qq908382818** [reply]
not the same question,

**Reply by stevemonaco** [reply]
It's the same material. System.Drawing.Bitmap isn't crossplatform, so it can't be used in Avalonia. The discussion shows code how to copy the `Mat` into a `WriteableBitmap`.

**Reply by qq908382818** [reply]
not work,why? 
 How to implement this method?
avalonia v11  net 8.0 thanks! give me code
    public static WriteableBitmap ToWriteableBitmap(OpenCvSharp.Mat mat);

**Comment by qq908382818** [comment]
public static WriteableBitmap ToWriteableBitmap2(Mat mat)
 {
     var bitmap = new WriteableBitmap(
         new PixelSize(mat.Width, mat.Height),
         new Vector(mat.Width, mat.Height),
         PixelFormat.Bgra8888);

     using (var buffer = bitmap.Lock())
     {
         mat.GetArray(out byte[] bufferMat);//OpenCvSharp.OpenCvSharpException:“Mat data type is not compatible: CV_8UC3”
         Marshal.Copy(bufferMat, 0, buffer.Address, bufferMat.Length);
     }

     return bitmap;
 }
error:OpenCvSharp.OpenCvSharpException:“Mat data type is not compatible: CV_8UC3”

**Reply by stevemonaco** [reply]
I don't understand why you aren't following the discussion, particularly this comment: https://github.com/AvaloniaUI/Avalonia/discussions/11074#discussioncomment-5777269

Besides that, you can't directly copy here. Your image is 32bit RGBA (Bgra8888) and your OpenCV type is 24bit RGB (CV_8UC3). They vary by an alpha channel.
